blocks/product-support/product-support.js

```diff
@@ -0,0 +1,152 @@
+import { createDomStructure, translate } from '../../scripts/lib-franklin.js';
+
+function getInfo() {
+  const url = new URL(window.location);
+  const idx = url.pathname.indexOf('/product-support/');
+  if (idx > 0) {
+    const slug = url.pathname.substring(url.pathname.indexOf('/product-support/') + '/product-support/'.length);
+    if (slug) {
+      return {
+        productCode: slug,
+        productSupport: url.pathname.substring(0, url.pathname.indexOf(`${slug}`)),
+        language: url.pathname.substring(1, url.pathname.indexOf('/', 1)),
+      };
+    }
+  }
+  return { productSupport: url.pathname.substring(0, url.pathname.length - 1) };
+}
+
+function getProduct(json, productCode, language) {
+  const product = json.Product.data
+    .find((entry) => entry.ProductCodes.split('|')
+      .some((code) => code === productCode)
+        && entry.Languages.split('|')
+          .some((productLanguage) => productLanguage.toUpperCase() === language.toUpperCase()));
+
+  if (product) {
+    const translation = json.ProductTranslation.data
+      .find((entry) => entry.ProductRef === product.ProductCodes
+        && entry.Language === language);
+    if (translation) {
+      if (translation.Name) {
+        product.Name = translation.Name;
+      }
+      if (translation.Image) {
+        product.Image = translation.Image;
+      }
+    }
+  }
+
+  return product;
+}
+
+function getTypes(json, product, language) {
+  return json.ProductAsset.data
+    .filter((asset) => asset.ProductRef === product.ProductCodes && asset.Languages
+      .split('|').some((assetLanguage) => assetLanguage.toUpperCase() === language.toUpperCase()))
+    .map((asset) => asset.Type)
+    .filter((value, index, array) => array.indexOf(value) === index);
+}
+
+function getAssets(json, product, language, type, allType) {
+  return json.ProductAsset.data
```

Comment:
if you liked the previous comment, we could make it more readable like this (also, i think `includes` is somehow more speaking than `some`, but that's just personal pref):

```
  const languageUpper = language.toUpperCase();

  return json.ProductAsset.data.filter((asset) => {
    const codeMatch = asset.ProductRef === product.ProductCodes;
    const langMatch = asset.Languages.split('|').map((lang) => lang.toUpperCase()).includes(languageUpper);
    const typeMatch = type === allType || asset.Type === type;

    return codeMatch && langMatch && typeMatch;
  });
```

Reply:
Done